OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -193,6 +247,7 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
                             fromCache = false
                         )
                     }
+                    resetPurchase()
```

Comment:
Just a thought => how about verifying manually is the course upgraded successfully or not.
Scenario: try to purchase the course then disconnect the internet and let the execute API & further flow to be completed n check

Reply:
To remove any edge cases in process of configuration change on `FullscreenLoaderDialogFragment` we have made this dialog `nullable`. So we won't call the loader dialog in any unwanted situation.